addons/account/models/reconciliation_widget.py

```diff
@@ -0,0 +1,782 @@
+                is_partner and ' ' or "AND at.type <> 'payable' AND at.type <> 'receivable'",
+                account_type and "AND at.type = %(account_type)s" or '',
+                res_ids and 'AND ' + res_alias + '.id in %(res_ids)s' or '',
+                self.env.user.company_id.id,
+                is_partner and 'AND l.partner_id = p.id' or ' ',
+                is_partner and 'l.partner_id, p.id,' or ' ',
+                res_alias=res_alias
+            ))
+        self.env.cr.execute(query, locals())
+
+        # Apply ir_rules by filtering out
+        rows = self.env.cr.dictfetchall()
+        ids = [x['account_id'] for x in rows]
+        allowed_ids = set(Account.browse(ids).ids)
+        rows = [row for row in rows if row['account_id'] in allowed_ids]
+        if is_partner:
+            ids = [x['partner_id'] for x in rows]
+            allowed_ids = set(Partner.browse(ids).ids)
+            rows = [row for row in rows if row['partner_id'] in allowed_ids]
+
+        # Fetch other data
+        for row in rows:
+            account = Account.browse(row['account_id'])
+            row['currency_id'] = account.currency_id.id or account.company_id.currency_id.id
+            partner_id = is_partner and row['partner_id'] or None
+            row['reconciliation_proposition'] = self._get_move_line_reconciliation_proposition(account.id, partner_id)
+        return rows
+
+    @api.model
+    # model: 'account.move.line',
+    # method: 'process_reconciliations',
+    def process_move_lines(self, move_line_ids, data):
+        """ Used to validate a batch of reconciliations in a single call
+            :param data: list of dicts containing:
+                - 'type': either 'partner' or 'account'
+                - 'id': id of the affected res.partner or account.account
+                - 'mv_line_ids': ids of existing account.move.line to reconcile
+                - 'new_mv_line_dicts': list of dicts containing values suitable for account_move_line.create()
+        """
+
+        Partner = self.env['res.partner']
+        Account = self.env['account.account']
+
+        for datum in data:
+            if len(datum['mv_line_ids']) >= 1 or len(datum['mv_line_ids']) + len(datum['new_mv_line_dicts']) >= 2:
+                self._process_move_lines(datum['mv_line_ids'], datum['new_mv_line_dicts'])
+
+            if datum['type'] == 'partner':
+                partners = Partner.browse(datum['id'])
+                partners.mark_as_reconciled()
+            if datum['type'] == 'account':
+                accounts = Account.browse(datum['id'])
+                accounts.mark_as_reconciled()
+
+    ####################################################
+    # Private
+    ####################################################
+
+    @api.model
+    # get_move_lines_for_reconciliation
+    def _get_move_lines(self, aml_accounts, partner_id, excluded_ids=None, search_str=False, offset=0, limit=None):
```

Comment:
Can you motivate the removal of the parameters overlook_partner and additional_domain?

Reply:
I made a modification, to have a method returning the domain. additional_domain and overlook_partner are never used in master or enterprise. If a community module needs to use it, just make an inheritance.

Reply:
overlook_partner is also use by _get_common_sql_query, I did not change this method, but it was never use in the get_move_lines_for_reconciliation method calls.